File: leetcode/easy/design-parking-system/java/Design-Parking-System.java
// Solution for design-parking-system (Java)
// Problem: https://leetcode.com/problems/design-parking-system/

class parkingSystem{
    private int[] slots;

    public ParkingSystem(int big, int medium, int small){
        slots = new int[]{big, medium, small};
    }

    public boolean addCar(int carType){
        if(slots[carType - 1] > 0){
            slots[carType - 1]--;
            return true;
        }
        return false;
    }
}

File: leetcode/easy/design-parking-system/python/design-parking-system.py
# Solution for design-parking-system (Python)
# Problem: https://leetcode.com/problems/design-parking-system/

class ParkingSystem(object):

    def __init__(self, big, medium, small):
        """
        :type big: int
        :type medium: int
        :type small: int
        """
        self.spaces = {
            1: big, 
            2: medium, 
            3: small
        }
        

    def addCar(self, carType):
        """
        :type carType: int
        :rtype: bool
        """
        if self.spaces[carType] > 0:
            self.spaces[carType] -= 1
            return True
        return False
        